update my fork's master

suresttexas00 opened the pull request.

JasonBristol commented:
I have also fixed imports that were lost after #4

suresttexas00 commented on src/globals.py:
```diff
@@ -1,2 +1,4 @@
+
 build = 111
-type = 'dev'
+type = 'dev' # type is a reserved word, we should consider changing this
```
definitely.  This is very common in wrapper code - using reserved words for local variables or shadowing names from an outer scope.

JasonBristol replied:
I addressed many of these in #18 however a few still linger, I will spin up an issue to track this

suresttexas00 commented on src/proxy.py:
```diff
@@ -667,29 +661,31 @@ def argsAfter(i):
                                                    "action": "finish_using"}):
                         return False
-            if self.server.state != 3:
+            if not self.server.state == 3:
```
I know this is mostly a matter of personal preference in regards to readability, but I think the community consensus is that the `if cond != value` is preferred and also has a (very slight) speed advantage.

http://stackoverflow.com/questions/31026754/python-if-not-vs-if

JasonBristol replied:
That is a good point, I think that was an accident when I was adjusting a lot of conditional logic, I will revert that since I think that is the only instance where that occured

suresttexas00 commented on src/web.py:
```diff
@@ -14,6 +14,7 @@
 import os
 import md5
```
md5 is deprecated.  We should probably upgrade this to hashlib

JasonBristol commented:
I also port to xrange() for iterative loops since this typically has better performance in python 2, we would want to backport this later for any python 3 work since xrange() no longer exists and its benefit has been rolled into range()

suresttexas00 commented on src/__main__.py:
```diff
@@ -163,22 +160,21 @@ def lookupUsernamebyUUID(self, useruuid):
             self.usercache[useruuid] = {"time": time.time(), "original": None, "name": None,
                                         "online": True, "localname": None, "IP": None, "names": []}
-        for i in range(0, numbofnames):
+        for i in xrange(0, numbofnames):
```
I'm concerned here because xrange is gone from python 3.. should we upgrade the code.. Is there a performance reason for this? 

http://stackoverflow.com/questions/135041/should-you-always-favor-xrange-over-range

JasonBristol replied:
Since we are not using range() to build a list and are only using it to iterate we get a performance boost (if I recall it is significant). We can backport python 3's range() through the __future__ module, but in the essence of a quick value add I ported to xrange() until we start to seriously consider py3

suresttexas00 commented:
OK.. if you are done for now, I will Merge this.

JasonBristol commented:
This should be set for merge now

suresttexas00 committed:
edit minecraft.py documents

suresttexas00 committed:
edit minecraft.py documents

suresttexas00 committed:
Merge pull request #419 from suresttexas00/development

Edit minecraft.py documentation.

suresttexas00 committed:
edit minecraft.py documents

suresttexas00 committed:
Merge pull request #420 from suresttexas00/development

edit minecraft.py documents (fix minor layout error)

suresttexas00 committed:
edit docstrings and clean code (primarily player.py)

suresttexas00 committed:
edit docstrings for base.py

suresttexas00 committed:
edit docstrings for minecraft.py

suresttexas00 committed:
edit docstrings for minecraft.py

suresttexas00 committed:
edit docstrings for minecraft.py

suresttexas00 committed:
finish editing docstrings for minecraft.py

suresttexas00 committed:
editing docstrings for player.py

suresttexas00 committed:
editing docstrings for player.py

suresttexas00 committed:
final editing docstrings for player.py

suresttexas00 committed:
editing docstrings World and entity

suresttexas00 committed:
finish editing docstrings World and entity

suresttexas00 committed:
finish editing docstrings.  Make readout public to helpers api

suresttexas00 committed:
finish editing docstrings.  Make readout public to helpers api

suresttexas00 committed:
Merge pull request #421 from suresttexas00/development

Finish making nice API docstrings for Documentation.

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
a few touches and typos

suresttexas00 committed:
Think I am now finished with the doc strings

suresttexas00 committed:
Merge pull request #422 from suresttexas00/development

Make documentation look even better.  Add more info to Class descriptions

suresttexas00 committed:
improve wrapperconfig doc header

suresttexas00 committed:
Merge pull request #423 from suresttexas00/development

improve wrapperconfig doc header

suresttexas00 committed:
more doc improvements before starting on event documenter

suresttexas00 committed:
Merge pull request #424 from suresttexas00/development

more doc improvements before starting on event documenter

suresttexas00 committed:
first event builder commit

suresttexas00 committed:
test event builder commit

suresttexas00 committed:
alpha of event builder

suresttexas00 committed:
Merge pull request #425 from suresttexas00/development

Start Event documentation creator.

suresttexas00 committed:
better event documenter to replace crappy one I wrote late at night

suresttexas00 committed:
Merge pull request #426 from suresttexas00/development

better event documenter to replace crappy one I wrote late at night

suresttexas00 committed:
fix typos, try out new docstring processor

suresttexas00 committed:
Merge pull request #427 from suresttexas00/development

fix typos, try out new docstring processor

suresttexas00 committed:
Update index.md

suresttexas00 committed:
Update index.md

suresttexas00 committed:
Merge pull request #4 from benbaptist/development

Development

suresttexas00 committed:
Merge pull request #428 from suresttexas00/development

Merge pull request #4 from benbaptist/development

suresttexas00 committed:
Fix Storage pickling so it is more reliable

suresttexas00 committed:
Build 197, Version 0.11.2
- Build 197:
    - Fix load_json and load_pickle functions which returned the values
      without actually updating the Data dictionary.
    - Clean up wrapper `__main__` code and add argument parsing to
      wrapper startup.  Added '--betterconsole' and '--encoding'.
    - changed build script to make incrementing builds an option.

suresttexas00 committed:
correct build number

suresttexas00 committed:
Merge pull request #429 from suresttexas00/development

Build 196 - fix storages

suresttexas00 committed:
build 197 fix issue #430

suresttexas00 committed:
Merge pull request #431 from suresttexas00/development

fix issue #430

Katipo007 committed:
Lvl 4 OPs bypass api.registerCommand() permissions

Expanded if statement on line 117 to allow sever OPs with level 4 or higher to execute commands defined by api.registerCommand() regardless of permissions.

OPs of this level have the ability to give themselves any permissions anyway, why not save them the time of giving themselves all the extra permissions?

suresttexas00 committed:
Merge pull request #432 from Katipo007/patch-1

Lvl 4 OPs bypass api.registerCommand() permissions

suresttexas00 committed:
Merge pull request #5 from benbaptist/development

Development pull changes from  Katipo007

suresttexas00 committed:
ensure only OP level 4 has permissions commands

suresttexas00 committed:
Merge pull request #434 from suresttexas00/development

Ensure only OPs 4 have ability to set permissions

suresttexas00 committed:
stage new permissions module and new isOp() levels

suresttexas00 committed:
few corrections, additions, typos...

suresttexas00 committed:
Merge pull request #436 from suresttexas00/development

Initial staging of new permissions

suresttexas00 committed:
debugged and tested permissions.py code

suresttexas00 committed:
Merge pull request #438 from suresttexas00/development

debugged and tested permissions.py code

suresttexas00 committed:
build 199: implement new permissions and OPs code

suresttexas00 committed:
Fix various bugs with new Permissions
1) Fix log formatting errors in Storage.
2) change player (uuid) login data storage to Pickle due to Py3 errors with
 json not able to order the time items. We dont read that anyway...
3) remove test code in permissions (buggy in production setting).
4) Work bugs out of 'info' commands in /perms.
5) Correct bugs with getting superOP name/levels.
6) Documentation updates.

suresttexas00 committed:
Merge pull request #439 from suresttexas00/development

New permissions system

suresttexas00 committed:
Update README.md

suresttexas00 committed:
Merge pull request #6 from benbaptist/development

Development

suresttexas00 committed:
change event document link in docs (api.base)

suresttexas00 committed:
Merge branch 'development' of https://github.com/suresttexas00/minecraft-wrapper into development

suresttexas00 committed:
Merge pull request #440 from suresttexas00/development

Update docs

suresttexas00 committed:
Remove grahamedgecombe.com from code for entities and blocks
The grahamedgecombe.com service we were using for entity definintions
is not playing nicely with Python requests (causes it to hang).  I
pulled the grahamedgecombe.com stuff out of our code, created a new
copy and made it compatible with the new 1.11

suresttexas00 committed:
Merge pull request #441 from suresttexas00/development

Remove grahamedgecombe.com from code for entities and blocks

suresttexas00 committed:
correct permissions problem with registered perrmissions

suresttexas00 committed:
Merge pull request #442 from suresttexas00/development

correct permissions problem with registered perrmissions

suresttexas00 committed:
Bugfixes build 201
- Bugfix registered_permissions.
- Bugfix group_delete method.
- Bugfix entity renaming (was using pre-1.11 in call cases).
- Correct bug in hasPermission that would set False the first time
  a user permission was checked (even if it was a default or None)
- Make help readout better.  Display the commands (but grayed out)
  so that users don't get blank help pages when they have no
  available commands on that page.

suresttexas00 committed:
Merge pull request #444 from suresttexas00/development

Bugfixes build 201

suresttexas00 committed:
fix event timer bug causing backups (or any timed event) not to run,

suresttexas00 committed:
Merge pull request #446 from suresttexas00/development

fix event timer bug causing backups (or any timed event) not to run,

suresttexas00 committed:
Version 0.11.5, build 203
- Add wrapper "op" command Feature Request:
    - Added two API minecraft items to makeOP() and deOp()
    - console and in-game command connections all use wrapper's
      op/deop API.  See documention in api minecraft.
- various bug fixes
    - fix set_item to allow adding new entries to a file.
    - tweaks here and there: converting to new classes, removing default
     '-r' arguments for file reading, etc.
    - clarify in code that what we called 'operatordict' in mcserver.py
     is actually a list (and is now renamed 'operator_list').
    - restore spawn object parsing (seems harmless since objects
     are GC-ed as entities)
    - remove some old debug and print statements.
    - Fix small error in in-game /help command.

suresttexas00 committed:
Merge pull request #447 from suresttexas00/development

Version 0.11.5, build 203

suresttexas00 committed:
update readchar package and use new classes in code

suresttexas00 committed:
Merge pull request #448 from suresttexas00/development

update readchar package and use new classes in code

suresttexas00 committed:
insubstantial changes (gitignore changes)

suresttexas00 committed:
Merge pull request #449 from suresttexas00/development

insubstantial changes (gitignore changes)

suresttexas00 committed:
updates to ReadChar package

suresttexas00 committed:
Merge pull request #450 from suresttexas00/development

updates to ReadChar package

suresttexas00 committed:
add UP/DOWN arrow keys with history to console

suresttexas00 committed:
build 208 anti-scroll off improvements

suresttexas00 committed:
build 208 anti-scroll off improvements (typos)

suresttexas00 committed:
removed unused variables

suresttexas00 committed:
Merge pull request #452 from suresttexas00/development

build 208 anti-scroll off improvements

suresttexas00 committed:
improve helpers testing and some minor refactors.

suresttexas00 committed:
re-do wrapper halt signal system and move encryption package

suresttexas00 committed:
first strip of wrapper references from proxy

suresttexas00 committed:
found error that went unnoticed for a long time

suresttexas00 committed:
summarize changes.. something is wrong

suresttexas00 committed:
revert changes to mcserver.javaserver.protocolVersion

suresttexas00 committed:
total remodeling of proxy

suresttexas00 committed:
first alpha of refactor

suresttexas00 committed:
Debug to beta quality. player object api needs work.

suresttexas00 committed:
update packet information to minecraft 1.12

suresttexas00 committed:
changed mcserver self.vitals.players to self.wrapper.players

suresttexas00 committed:
version 0.12.1 debug server vitals system

suresttexas00 committed:
fix player payload error

suresttexas00 committed:
player object now fixed

suresttexas00 committed:
reorganize plugins.  Prepping for release

suresttexas00 committed:
debugging 1.12 and new servervitals

suresttexas00 committed:
build 218, stable plugin debugging

suresttexas00 committed:
fix tp in plugins.  Redo proxy constants

suresttexas00 committed:
build 219

suresttexas00 committed:
build 219 (working towards release)

suresttexas00 committed:
build 219 - lots of fixes!

suresttexas00 committed:
Merge pull request #454 from suresttexas00/development

Build 219, version 0.12.1,  Update to Minecraft 1.12

suresttexas00 committed:
update documentation

suresttexas00 committed:
Merge pull request #455 from suresttexas00/development

update documentation

suresttexas00 committed:
update documentation

suresttexas00 committed:
Merge pull request #456 from suresttexas00/development

update documentation (correct typo)

suresttexas00 committed:
small edits to teleports plugin

suresttexas00 committed:
Merge pull request #457 from suresttexas00/development

small edits to teleports plugin

suresttexas00 committed:
update of commands.py - last update for MC 12

suresttexas00 committed:
Merge pull request #458 from suresttexas00/development

update of commands.py - last update for MC 12

suresttexas00 committed:
update to Minecraft 1.12.1

suresttexas00 committed:
Merge pull request #459 from suresttexas00/development

update to Minecraft 1.12.1

suresttexas00 committed:
build 223

suresttexas00 committed:
correct small typo in wrapper/api/base.py

suresttexas00 committed:
reversed handle conditions to ensure parsing only attemped in play/lobby modes

suresttexas00 committed:
build 224, fix document generation some

suresttexas00 committed:
remove duplicated items (I hope)

suresttexas00 committed:
Merge pull request #463 from suresttexas00/development

build 224

suresttexas00 committed:
fix error I created in handles

suresttexas00 committed:
bugfix packet handles

suresttexas00 committed:
Merge pull request #464 from suresttexas00/development

Bug fix error in handle() methods

suresttexas00 committed:
Bugfix keepalives for 1.12.2

suresttexas00 committed:
Test build restricting position updates to CB PPosLook

suresttexas00 committed:
Test build2 restricting position updates to CB PPosLook

suresttexas00 committed:
Test build3 - implement parser in to CB PLAYER_POSLOOK packet

suresttexas00 committed:
Build 228, version 0.13.4- debug for MC1.12.2

suresttexas00 committed:
Merge pull request #466 from suresttexas00/development

Build 228, version 0.13.4- MC version 1.12.2 Updates

suresttexas00 committed:
Create About.md

Preparing to update Master (and maintain compatibility with old master version)

suresttexas00 committed:
Update and rename About.md to readme.md

suresttexas00 committed:
Update readme.md

suresttexas00 committed:
prepare for updating master (maintain updates compat.)

suresttexas00 committed:
staging for master branch update (backwards compat.)

suresttexas00 committed:
Merge pull request #467 from suresttexas00/development

Patching the old "Docs" folder back into repo

suresttexas00 committed:
bugfix in SB parse

suresttexas00 committed:
bugfix in SB parse

suresttexas00 committed:
bugfix in SB parse (tested and completed)

suresttexas00 committed:
Merge pull request #468 from suresttexas00/development

Bugfix 0.13.4

suresttexas00 committed:
Update master branch to current development
This is very overdue as the master branch has been out of date since minecraft 1.9.  The original master is now located in the "Original" branch.

suresttexas00 committed:
Merge pull request #469 from benbaptist/development

Development pull back to master.

suresttexas00 committed:
Merge pull request #8 from benbaptist/development

Development

suresttexas00 committed:
Update version.json

suresttexas00 committed:
Stable build

suresttexas00 committed:
Update version.json

Make changes visible to existing stable wrappers.

suresttexas00 committed:
Update version.json

Make changes visible to existing wrappers.

suresttexas00 committed:
Stable build readme update

suresttexas00 committed:
Merge pull request #470 from benbaptist/master

re-sync master/dev

suresttexas00 committed:
Merge pull request #9 from benbaptist/development

Development

suresttexas00 committed:
mark repo as dev again

suresttexas00 committed:
Merge pull request #471 from suresttexas00/development

Re-mark as dev repo

suresttexas00 committed:
start debuggin possible problems with keepalives

suresttexas00 committed:
start debuggin possible problems with keepalives

suresttexas00 committed:
continuing work with keepalives

suresttexas00 committed:
continuing work with keepalives

suresttexas00 committed:
improved keepalive handling

suresttexas00 committed:
Merge pull request #472 from suresttexas00/development

Fix keep alives for improved connections.

suresttexas00 committed:
Merge pull request #473 from benbaptist/development

Build 230 stable

suresttexas00 committed:
restore dev status after a stable push, increment build number.

suresttexas00 committed:
Merge pull request #474 from suresttexas00/development

restore dev status after a stable push, increment build number.

suresttexas00 committed:
clean up documentation and correct minor parse errors.

suresttexas00 committed:
fix error in playerstats due to trying to read pkl as json

suresttexas00 committed:
implement Pickling into helpers. debug playerstats.

suresttexas00 committed:
correct error in processcolorcodes

suresttexas00 committed:
correct error in playerstats

suresttexas00 committed:
changes from production version (oops)

suresttexas00 committed:
various updates to commands, espec. perms

suresttexas00 committed:
Official Build 232

suresttexas00 committed:
Merge pull request #475 from suresttexas00/development

Build 232 development

suresttexas00 committed:
Document cleanups

suresttexas00 committed:
Stable 0.13.5

suresttexas00 committed:
Merge pull request #476 from suresttexas00/development

Build 233, version 0.13.5

suresttexas00 committed:
Merge pull request #477 from benbaptist/development

Merge Development into master (0.13.5), build 233

suresttexas00 committed:
rebuild as dev

suresttexas00 committed:
Merge pull request #478 from suresttexas00/development

rebuild as dev

suresttexas00 committed:
Start ubil 234.  debug PlayerPosLook (SB)

suresttexas00 committed:
bugfix for Spigot servers

suresttexas00 committed:
Merge pull request #480 from suresttexas00/development

Bugfix for Spigot servers

suresttexas00 committed:
Merge pull request #481 from benbaptist/development

Development merge Spigot bugfix to master

suresttexas00 committed:
rebuild with dev status after stable push

suresttexas00 committed:
SPigot fixes and API improvements for non-proxy mode

suresttexas00 committed:
SPigot fixes and API improvements for non-proxy mode

suresttexas00 committed:
build #238
-refactoring proxy again:
    - removed "lobby" references from serverconnection: "LOBBY" could
    never be a real state for server connection since LOBBY is not a real
    server.  It is just a state in which client connection is active with
    NO current server.  clientconnection.py is the valid place to
    handle "lobby" states.
    - clean up clientconnection references to closing server connections
    - clean up serverconnection and parse_cb to remove old unused stuff.
    - turn off proxy entity filtering when entity controls are disabled.
-fix new bug from bugfix for Spigot in build #237 that introduced new
error preventing vanilla proxy server connections.

suresttexas00 committed:
correct typos

suresttexas00 committed:
second build 238
    - remove logging/console output for "PING REQUEST" connections.
    - refactor/rename client methods to be more/better descriptive.  For
    instance, client_logon() to Logon_client_into_proxy() to show that
    the client has not logged into a server yet, just into the wrapper proxy.
    - clean up clientconnection and parse_sb to remove unused stuff.
    - clean up the program flow for disconnections in serverconnection and
    clientconnection.
    - fix keepalive process to stop as soon as client aborts.  Ensure
    a prompt abort signal for terminated connections.
    - make normal debug messages look more like prose and less "scary".
    reserve full bebug style messages for unexpected items/errors.
    - improve player/client logout process to clean player and client
    objects out reliably.  Client is only removed when the client proxy
    flush loop ends and the player object is removed at the mcserver.py
    logout().
    - re-add parse_play_player_position to parse_sb.py
    - re-implement "silent-ipbans" option.
    - beef up chat_to_server() method (the backend for api.player say() and
    execute() methods) to prevent a chat message being sent over 100 or 256
    characters, as applicable to the minecraft version.
    - added a chat_to_client() method to clientconnection.py as the
    preferred backend for player.message() when in proxy mode.

suresttexas00 committed:
change documentation index to readme.md

suresttexas00 committed:
Version 0.13.7
bump version
fix bug in player.message() I introduced in build 238

suresttexas00 committed:
small doc updates

suresttexas00 committed:
small doc updates

suresttexas00 committed:
build 240- more cleanup for PING requests.

suresttexas00 committed:
build 240- more cleanup for PING requests.

suresttexas00 committed:
fix chat_to_client() so that json is messaged properly.

suresttexas00 committed:
fix hardcoded update paths

suresttexas00 committed:
Merge branch 'development' of https://github.com/suresttexas00/minecraft-wrapper into development

suresttexas00 committed:
production updates

suresttexas00 committed:
build 241 with corrections to updates

suresttexas00 committed:
build 242. important player.message and wrapper update fixes

suresttexas00 committed:
test build

suresttexas00 committed:
fix core/wrapper update

suresttexas00 committed:
test build 2

suresttexas00 committed:
test build 3

suresttexas00 committed:
test build 4

suresttexas00 committed:
test build 6

suresttexas00 committed:
test build 7

suresttexas00 committed:
Version 0.14.0 - critical \update-wrapper bugfix

suresttexas00 committed:
Version 0.14.0 build 244 - critical \update-wrapper bugfix

suresttexas00 committed:
Build 245 [0.14.1] - Master branch update
- fix spigot login position (due to spigot pre-pending the world name to
 the player coordinates).
- while at that, add IP address to player object from login text (so that
 non-proxy wrapper's have the player IP address).
- build a master branch with dev fixes after build 239 (especially due to
 broken wrapper update stuff).
- (finally) fix chat_to_client() so that json is messaged properly.
- fix update code in core/wrapper.  changed the config items slightly.
- expanded capabilities of player message to send messages above XP bar (proxy mode)
- fix chat_to_client() so that json is messaged properly.  Past couple
builds broke advanced json chat things (like links and hoverevents).

suresttexas00 committed:
Merge pull request #490 from suresttexas00/development

Build 245 [0.14.1] - Master branch update

suresttexas00 committed:
continue Dev status after master update

suresttexas00 committed:
push changes to readme.  Need to edit README.md online

suresttexas00 committed:
Update README.md

suresttexas00 committed:
Update README.md

suresttexas00 committed:
push changes to readme.  Need to edit README.md online

suresttexas00 committed:
push changes to readme.

suresttexas00 committed:
Non-build commit staging new wrapper password handlers
- fix error in core/wrapper.py that causes spamming of "Disabling proxy
 mode because ..." to the console.
- added text to the vanilla message to tell console user not only the
 port of the minecraft server, but the proxy too (if proxy is enabled).
- added module `wrapper/utils/crypt.py` with full featured password encryption schemas:
    - `make_hash(password_string, encoding='utf-8')` - hash a password
    - `check_pw(password_string, hashed_pw_str, encoding='utf-8')` check password against hash
    - `phrase_to_url_safebytes(pass_phrase)` - create bytes for Fernet encryption from wrapper pass phrase
    - `encrypt(passphrase, data)` - do a Fernet encrypt of data from bytes pass phrase
    - `decrypt(passphrase, encrypted_str_data)` - Fernet decryption of data
- implement a wrapper password to be used internally by wrapper to
 store sensitive data (email passwords) that wrapper's features may use.
- implement encryption hashing of other stored passwords to disk
 (web, IRC, so forth.. ).

suresttexas00 committed:
(non-build commit)
- corrected Web/dashboard and IRC password references to use hashed
 passwords.

suresttexas00 committed:
Build 246 [0.14.1] Dev
- fix error in core/wrapper.py that causes spamming of "Disabling proxy
 mode because ..." to the console.
- added text to the vanilla message to tell console user not only the
 port of the minecraft server, but the proxy too (if proxy is enabled).
- added module `wrapper/utils/crypt.py` with full featured password encryption schemas:
    - `make_hash(password_string, encoding='utf-8')` - hash a password
    - `check_pw(password_string, hashed_pw_str, encoding='utf-8')` check password against hash
    - `phrase_to_url_safebytes(pass_phrase)` - create bytes for Fernet encryption from wrapper pass phrase
    - `encrypt(passphrase, data)` - do a Fernet encrypt of data from bytes pass phrase
    - `decrypt(passphrase, encrypted_str_data)` - Fernet decryption of data
- implement a wrapper password to be used internally by wrapper to
 store sensitive data (email passwords) that wrapper's features may use.
- implement encryption hashing of other stored passwords to disk
 (web, IRC, so forth.. ).
- corrected Web/dashboard and IRC password references to use hashed
 passwords.
- Added /password command to wrapper to allow setting of encrypted
 passwords in the wrapper.config.json file.
** important note ** I may consider uniting make_hash/encrypt so that a single schema is used in the next build

suresttexas00 committed:
Build 247 [0.14.2] Dev
- For consistency and eliminate confusion, all wrapper internal
 encryption uses `cryptography` package even for regular hashing
 and password checking.
- retained bcrypt package as on option, but it will likely only ever
 be a part of the plugin API.
- added plugin API (base.py) password handler functions:
  ```
      def hash_password(self, password):
         """ Bcrypt-based password encryption.  Takes a raw string password
         returns a string representation of the binary hash."""

      def check_password(self, password, hashed_password):
         """ Bcrypt-based password checker.  Takes a raw string password and
         compares it to the hash of a previously hashed password, returning
         True if the passwords match, or False if not.
  ```

suresttexas00 committed:
redact version/build update due to serious error

suresttexas00 committed:
fixed readline/betterconsole error/halt

suresttexas00 committed:
fixed readline/betterconsole error

suresttexas00 committed:
fix server event timers

suresttexas00 committed:
build 248

suresttexas00 committed:
Build 249
- add config items to allow entering a plaintext password that
 gets digested and resaved with encryption.
- PEP-8 updates to dashboard (still not usable yet)
- Fix error that prevented startup if user did not have cryptography
 installed.

suresttexas00 committed:
build 250 - Fix custom messages again for non-proxy mode.

suresttexas00 committed:
bugfix password config setup code in wrapper.py

suresttexas00 committed:
apparently, Spigot/Mcbans needs more time to kick the players before a server restart...

suresttexas00 committed:
real build 251  for benbaptist development

suresttexas00 committed:
Build 252
- correct password decryting errors in IRC
- IRC now works in PY2 only (unicode related issues in PY3)

suresttexas00 committed:
build 253 - Fix IRC fro Python3

suresttexas00 committed:
Build 254 [0.14.2] - Master branch update
- Correct one other Python 2/3 error in IRC
- Add explanatory comments about player object usage in strings
- Add warning log about web mode being presently broken
- Build master branch to [0.14.2]

suresttexas00 committed:
correct a couple typos in readme.md

suresttexas00 committed:
Build 254 [0.14.2] - Master branch update
- Correct one other Python 2/3 error in IRC
- Add explanatory comments about player object usage in strings
- Add warning log about web mode being presently broken
- Build master branch to [0.14.2]

suresttexas00 committed:
Build 254 [0.14.2] - Master branch update
- Correct one other Python 2/3 error in IRC
- Add explanatory comments about player object usage in strings
- Add warning log about web mode being presently broken
- Build master branch to [0.14.2]

suresttexas00 committed:
Update to version 0.15.0
- added module `wrapper/utils/crypt.py` with full featured password encryption schemas:
```
    <Class Crypt>
    - `bcrypt_make_hash(password_string)` - hash a password
    - `bcrypt_check_pw(password_string, hashed_pw_str,)` check password against hash
    - `check_pw(password_string, hashed_pw_str)` Fernet password checker
    - `encrypt(data)` - do a Fernet encrypt of data.
    - `decrypt(encrypted_str_data)` - Fernet decryption of data
```
- implement a wrapper password to be used internally by wrapper to
 create the Fernet cipher used by the Crypt class.
- implement encryption hashing of all stored passwords to disk
 (web, IRC, so forth.. ).
- add config items to allow entering plaintext passwords that
 get digested and re-saved with Fernet encryption.
- added plugin API (base.py) non-Fernet password handler functions:
  ```
      def hash_password(self, password):
         """ Bcrypt-based password encryption.  Takes a raw string password
         returns a string representation of the binary hash."""

      def check_password(self, password, hashed_password):
         """ Bcrypt-based password checker.  Takes a raw string password and
         compares it to the hash of a previously hashed password, returning
         True if the passwords match, or False if not.
  ```
- Added /password command to wrapper to allow setting of encrypted
 passwords in the wrapper.config.json file.
- Bug fix [issue 492](https://github.com/benbaptist/minecraft-wrapper/issues/492):
    - Fix timer loops for reboots
    - Fix custom messages in Misc section and add new message "halt-message"
     for when wrapper is halted.
- Correct Python 3 errors in IRC (IRC now works in Python 2 and 3)
- Add explanatory comments about player object usage in strings
- Add warning log about web mode being presently broken
- fix error in core/wrapper.py that caused spamming of "Disabling proxy
 mode because ..." to the console.
- added text to the vanilla message to tell console user not only the
 port of the minecraft server, but the proxy too (if proxy is enabled).
- Few spigot fixes.

suresttexas00 committed:
Merge pull request #505 from suresttexas00/development

Build 254 [0.15.0] - Master branch update

suresttexas00 committed:
Create CONTRIBUTING.md

Initial Contrib pasted from wiki Tab because Github does not think we have contributer guidelines.. :(

suresttexas00 committed:
Stupid to create another file "just because" github suggested it

We already had this in the Wiki
